homeassistant/components/fritzbox/light.py

```diff
@@ -74,9 +74,6 @@ def brightness(self) -> int:
     @property
     def hs_color(self) -> tuple[float, float] | None:
         """Return the hs color value."""
-        if self.data.color_mode != COLOR_MODE:
-            return None
```

Comment:
Hm, there I don't remember why I did this in the first place, but it seems to work without it 😉

Reply:
had a discussion with Martin on discord about it - this is dead code (_will never be reached_), therefor it is not needed

Reply:
Are you sure? This comes from the library and the device can have either `color_mode=COLOR_TEMP_MODE` or `color_mode=COLOR_MODE`

Reply:
we already check the `self.data.color_mode` in the `color_mode` property. Based on the value of the `color_mode` property, either the `hs_color` or `color_temp_kelvin` property is accessed

Reply:
Ah, thanks for explaining. In that case I think we could remove this check in the `color_temp_kelvin` property too.

Reply:
when removing this check in `color_temp_kelvin` the test run into errors. because i do not own a fritz light bulb, i cannot test if there is something wrong with the mock in the tests or if it is a real issue.
nevertheless, i would liek to rework the `light` entities a bit to make use of the shorthand `_attr_xxx` attributes - but this will go into another PR

Reply:
Okay, lets merge it as it is now. Ping me when you have your PR ready and I can test it 😃